File: Two-Sum-Problem/python/twosum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        rems: Dict[str, int] = {}
        for i in range(len(nums)):
            key = target - nums[i]
            if str(nums[i]) in rems.keys():
                return [rems[str(target - key)], i]
            else:
                rems[str(key)] = i
        return [-1, -1]
  

File: Two-Sum-Problem/java/TwoSum.java
class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> table = new HashMap<>();
        for(int i=0; i< nums.length; i++){
            int rem = target - nums[i];
            if(table.containsKey(nums[i])){
                return new int[]{table.get(target - rem), i};
            }else{
                table.put(rem, i);
            }
        }
        return new int[]{-1,-1};
    }
}
